src/persist/src/indexed/metrics.rs

```diff
@@ -106,41 +154,52 @@ impl Metrics {
                 name: "mz_persist_cmd_run_ms",
                 help: "time spent running commands",
             )),
+            cmd_failed_count: registry.register_third_party_visible(metric!(
+                name: "mz_persist_cmd_failed_count",
+                help: "count of commands run where an error was returned",
+            )),
             cmd_step_ms: registry.register_third_party_visible(metric!(
                 name: "mz_persist_cmd_step_ms",
                 help: "time spent in step",
             )),
-            cmd_write_count: registry.register_third_party_visible(metric!(
-                name: "mz_persist_cmd_write_count",
-                help: "count of write commands run",
+            cmd_step_error_count: registry.register_third_party_visible(metric!(
+                name: "mz_persist_cmd_step_error_count",
+                help: "count of errors returned by cmd step",
+            )),
+            compaction_count: registry.register_third_party_visible(metric!(
+                name: "mz_persist_compaction_count",
+                help: "count of times unsealed and trace compaction occurred",
             )),
             compaction_ms: registry.register_third_party_visible(metric!(
                 name: "mz_persist_compaction_ms",
-                help: "time spent compacting trace and unsealed",
+                help: "time spent compacting unsealed and trace",
+            )),
+            compaction_write_bytes: registry.register_third_party_visible(metric!(
+                name: "mz_persist_compaction_bytes",
+                help: "bytes written compacting unsealed and trace",
+            )),
+            cmd_write_count: registry.register_third_party_visible(metric!(
+                name: "mz_persist_cmd_write_count",
+                help: "count of write commands run",
             )),
             cmd_write_record_count: registry.register_third_party_visible(metric!(
                 name: "mz_persist_cmd_write_record_count",
                 help: "total count of records written in all streams",
             )),
-            blob_write_count: registry.register_third_party_visible(metric!(
-                name: "mz_persist_blob_write_count",
-                help: "count of blob storage writes",
-            )),
-            blob_write_bytes: registry.register_third_party_visible(metric!(
-                name: "mz_persist_blob_write_bytes",
-                help: "total size written to blob storage",
+            cmd_write_record_bytes: registry.register_third_party_visible(metric!(
+                name: "mz_persist_cmd_write_record_bytes",
+                help: "total count of records written in all streams",
             )),
-            blob_write_ms: registry.register_third_party_visible(metric!(
-                name: "mz_persist_blob_write_ms",
-                help: "time spent writing to blob storage",
+            unsealed: BlobMetricsByType::register_with("unsealed", registry),
```

Comment:
An alternative to this is to make all the counters into UIntCounterVec with a label for each blob type, but my reading of the prometheus docs, suggested to me that they should just be individual counters. From https://docs.rs/prometheus/0.12.0/prometheus/struct.Opts.html#structfield.const_labels:

> const_labels are used to attach fixed labels to this metric. Metrics with the same fully-qualified name must have the same label names in their ConstLabels.
>
> Note that in most cases, labels have a value that varies during the lifetime of a process. Those labels are usually managed with a metric vector collector (like CounterVec, GaugeVec). ConstLabels serve only special purposes. One is for the special case where the value of a label does not change during the lifetime of a process, e.g. if the revision of the running binary is put into a label. Another, more advanced purpose is if more than one Collector needs to collect Metrics with the same fully-qualified name. In that case, those Metrics must differ in the values of their ConstLabels. See the Collector examples.
>
> If the value of a label never changes (not even between binaries), that label most likely should not be a label at all (but part of the metric name).